src/main/java/com/firebolt/jdbc/service/FireboltStatementService.java

```
@@ -85,6 +86,32 @@ public String executeAsyncStatement(StatementInfoWrapper statementInfoWrapper,
 		}
     }
 
+	/**
+	 * Executes statement with files
+	 *
+	 * @param statementInfoWrapper the statement info
+	 * @param properties the connection properties
+	 * @param statement the statement
+	 * @param files map of file identifiers to file contents
+	 * @return an Optional ResultSet if the statement returns results
+	 */
+	public Optional<ResultSet> executeWithFiles(StatementInfoWrapper statementInfoWrapper,
+												 FireboltProperties properties, FireboltStatement statement,
+												 Map<String, byte[]> files)
+			throws SQLException {
+		int queryTimeout = statement.getQueryTimeout();
+		InputStream is = statementClient.executeSqlStatementWithFiles(statementInfoWrapper, properties, queryTimeout, false, files);
+		if (statementInfoWrapper.getType() == StatementType.QUERY) {
+			return Optional.of(createResultSet(is, (QueryRawStatement) statementInfoWrapper.getInitialStatement(), properties, statement));
+		} else {
+			// If the statement is not a query, read all bytes from the input stream and close it.
```

let's throw an exception. Otherwise it will look like the call was successful.

---

This is the expected behavior since only a query type statement (SELECT/SHOW/DESCRIBE) would return a ResultSet. This method's return acts like the execute method. It should return something (true in the case of execute method) if there is a result and nothing (optional empty here or false for execute method) if it's a non-query. I would say to leave this as is since the ParquetStatement class will throw an error if a query is executed with executeUpdate. If a query is ran on execute, than the user can see it via the returned boolean value